File: peek_plugin_diagram/plugin-module/_private/tuples/DiagramTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {diagramTuplePrefix} from "../PluginNames";


@addTupleType
export class DiagramTuple extends Tuple {
    public static readonly tupleName = diagramTuplePrefix + "DiagramTuple";

    //  Description of date1
    dict1 : {};

    //  Description of array1
    array1 : any[];

    //  Description of date1
    date1 : Date;

    constructor() {
        super(DiagramTuple.tupleName)
    }
}

File: peek_plugin_diagram/_private/tuples/DiagramTuple.py
from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_diagram._private.PluginNames import diagramTuplePrefix


@addTupleType
class DiagramTuple(Tuple):
    """ Diagram Tuple

    This tuple is a create example of defining classes to work with our data.
    """
    __tupleType__ = diagramTuplePrefix + 'DiagramTuple'

    #:  Description of date1
    dict1 = TupleField(defaultValue=dict)

    #:  Description of date1
    array1 = TupleField(defaultValue=list)

    #:  Description of date1
    date1 = TupleField()
